jedi/api/__init__.py

```diff
@@ -108,7 +110,11 @@ def __init__(self, source=None, line=None, column=None, path=None,
         self._user_context = UserContext(self.source, self._pos)
         self._parser = UserContextParser(self._grammar, self.source, path,
                                          self._pos, self._user_context)
-        self._evaluator = Evaluator(self._grammar)
+        if sys_path is None:
+            venv = os.getenv('VIRTUAL_ENV')
+            if venv:
+                sys_path = list(get_venv_path(venv)) + sys.path
```

Comment:
However that `VIRTUAL_ENV` variable will not change, just because you're activating a virtualenv. So this logic effectively doesn't do anything else than loading the same `sys.path`?!

Or are you thinking about changing the environment variable before each Jedi Script invocation? In that case I would rather add another parameter to Script.